aten/src/ATen/native/cpu/GridSamplerKernel.cpp

```diff
@@ -764,6 +778,197 @@ struct ApplyGridSample<scalar_t, 2, GridSamplerInterpolation::Nearest,
   }
 };
 
+template<typename scalar_t, GridSamplerPadding padding, bool align_corners>
+struct ApplyGridSample<scalar_t, 2, GridSamplerInterpolation::Bicubic,
+                       padding, align_corners> {
+  using Vec = Vec256<scalar_t>;
+  using integer_t = int_same_size_t<scalar_t>;
+  using iVec = Vec256<integer_t>;
+
+  const int64_t inp_H;
+  const int64_t inp_W;
+  const int64_t inp_sH;
+  const int64_t inp_sW;
+  const int64_t C;
+  const int64_t inp_sC;
+  const ComputeLocation<scalar_t, padding, align_corners> compute_H;
+  const ComputeLocation<scalar_t, padding, align_corners> compute_W;
+  const bool must_in_bound = padding != GridSamplerPadding::Zeros;
+
+  ApplyGridSample(const TensorAccessor<scalar_t, 4>& input)
+    : inp_H(input.size(2))
+    , inp_W(input.size(3))
+    , inp_sH(input.stride(2))
+    , inp_sW(input.stride(3))
+    , C(input.size(1))
+    , inp_sC(input.stride(1))
+    , compute_H(input.size(2))
+    , compute_W(input.size(3)) {}
+
+  inline Vec cubic_convolution1(const Vec& tx, const Vec& A) const {
+    return ((A + Vec(2)) * tx - (A + Vec(3))) * tx * tx + Vec(1);
+  }
+
+  inline Vec cubic_convolution2(const Vec& tx, const Vec& A) const {
+    return ((A * tx - Vec(5) * A) * tx + Vec(8) * A) * tx - Vec(4) * A;
+  }
+
+  inline void get_cubic_coefficients(Vec coeffs[4], const Vec& tx) const {
```

Comment:
Should this (and below) be `Vec& coeffs`?

Reply:
Oh, I see, there are [two way to pass array to function](https://stackoverflow.com/questions/5724171/passing-an-array-by-reference)

`get_cubic_coefficients(Vec coeffs[])` would pass the pointer and lose the size information, use `get_cubic_coefficients(Vec (&coeffs)[4]` could pass the reference of array and maintain array information. Thanks!